### PR TITLE
8288495: [test] Make OutputAnalyzer exception more informative

### PR DESCRIPTION
**Pull request by RogerRiggs**
The RuntimeException from OutputAnalyzer for expected and unexpected exit values should include both the actual and expected values.
The method `shouldHaveExitvalue` and `shouldNotHaveExitValue` include the value expected and the value not expected instead of the actual value returned. The values provided in the exception are already known quantities in the code, what is not known is the actual value.  
The actual value is printed to stderr but not the expected value and in the logs, the exception report is frequently to stdout and therefor separated in the log from the output of stderr.  Including both expected and actual values in the exception makes it easier to understand.

Exception messages should not include newlines.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288495](https://bugs.openjdk.org/browse/JDK-8288495): [test] Make OutputAnalyzer exception more informative

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**) ⚠️ Review applies to [e73bf2bd](https://git.openjdk.org/jdk/pull/9247/files/e73bf2bd026022ce332be1f845750a8787b0a231)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [e73bf2bd](https://git.openjdk.org/jdk/pull/9247/files/e73bf2bd026022ce332be1f845750a8787b0a231)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9247/head:pull/9247` \
`$ git checkout pull/9247`

Update a local copy of the PR: \
`$ git checkout pull/9247` \
`$ git pull https://git.openjdk.org/jdk pull/9247/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9247`

View PR using the GUI difftool: \
`$ git pr show -t 9247`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9247.diff">https://git.openjdk.org/jdk/pull/9247.diff</a>

</details>

### CODE REVIEW

File: test/lib/jdk/test/lib/process/OutputAnalyzer.java [hunk 1]
```diff
@@ -502,7 +502,7 @@ public OutputAnalyzer shouldNotHaveExitValue(int notExpectedExitValue) {
         if (getExitValue() == notExpectedExitValue) {
             reportDiagnosticSummary();
             throw new RuntimeException("Unexpected to get exit value of ["
-                    + notExpectedExitValue + "]\n");
+                    + notExpectedExitValue + "], exit value is: [" + getExitValue() + "]");
```
Comment (naotoj) [inline test/lib/jdk/test/lib/process/OutputAnalyzer.java:505]: This seems redundant to me. "Unexpected to get exit value of [notExpectedExitValue]." should suffice?

Comment (dholmes-ora) [reply test/lib/jdk/test/lib/process/OutputAnalyzer.java:505]: I agree this doesn't need changing as there is only one value to report.

File: test/lib-test/jdk/test/lib/process/OutputAnalyzerTest.java [hunk 2]
```diff
@@ -216,6 +216,32 @@ public static void main(String args[]) throws Exception {
                 throw new Exception("firstMatch(String, int) failed to match. Expected: " + aa + " got: " + result);
             }
         }
+
+        {
+            try {
+                // Verify the exception message
+                OutputAnalyzer out = ProcessTools.executeProcess("true");
```
[the remark this answers is not in the document]
Comment (RogerRiggs) [reply test/lib-test/jdk/test/lib/process/OutputAnalyzerTest.java:223]: The Windows CreateProcess API appends ".exe".  (if there is no suffix).  
(And CI testing succeeded).

File: test/lib/jdk/test/lib/process/OutputAnalyzer.java (hunk 1 above)
Comment (RogerRiggs) [reply test/lib/jdk/test/lib/process/OutputAnalyzer.java:505]: Right, a bit overzealous on the expected vs actual.

**Review by lmesnik — APPROVED** [review]

**Comment by lmesnik** [comment]
Please update copyrights in the test file, otherwise, the last version looks good.

**Review by jaikiran — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]

**Pull request closed without merging**